src/Tuura/Boolean.hs

```diff
@@ -1,6 +1,9 @@
+{-# LANGUAGE FlexibleInstances, OverlappingInstances, IncoherentInstances #-}
```

Comment:
Yes, your current `Show` instance does not produce output that can be compiled. I suggest to use `derived Show` instance instead, and add a `prettyLiteral :: Literal String -> String` function.

By the way, if you'd like to avoid using names like `prettyCNF` and `prettyLiteral`, and have a single function instead, have a look at the standard `pretty` library. Specifically, there is a `Pretty` typeclass, which comes with `prettyShow` function that seems to do what you want:

https://hackage.haskell.org/package/pretty-1.1.3.5/docs/Text-PrettyPrint-Annotated-HughesPJClass.html